proxy/src/telemetry/metrics/histogram.rs

```diff
@@ -88,6 +88,98 @@ impl<V: Into<u64>> Histogram<V> {
         self.buckets[idx].incr();
         self.sum += value;
     }
+
+    // ===== Test-only methods to help with assertions about histograms. =====
+
+    /// Assert the bucket containing `le` has a count of at least `at_least`.
+    #[cfg(test)]
+    pub fn assert_bucket_at_least(&self, le: u64, at_least: u64) {
+        for (&bucket, &count) in self {
+            if bucket >= le {
+                let count: u64 = count.into();
+                assert!(
+                    count >= at_least,
+                    "le={:?}; bucket={:?};", le, bucket
+                );
+                break;
+            }
+        }
+    }
+
+    /// Assert the bucket containing `le` has a count of exactly `exactly`.
+    #[cfg(test)]
+    pub fn assert_bucket_exactly(&self, le: u64, exactly: u64) -> &Self {
+        for (&bucket, &count) in self {
+            if bucket >= le {
+                let count: u64 = count.into();
+                assert_eq!(
+                    count, exactly,
+                    "le={:?}; bucket={:?}; buckets={:#?};",
+                    le, bucket, self.buckets,
+                );
+                break;
+            }
+        }
+        self
+    }
+
+    /// Assert all buckets less than the one containing `value` have
+    /// counts of exactly `exactly`.
+    #[cfg(test)]
+    pub fn assert_lt_exactly(&self, value: u64, exactly: u64) -> &Self {
+        for (i, &bucket) in self.bounds.0.iter().enumerate() {
+            if let Bucket::Le(ceiling) = bucket {
```

Comment:
tioli: i find the big block and then `} else { break ]` to be a bit tricky to folow flow-control-wise.  Consider:

```rust
let ceiling = match bucket {
  Bucket::Le(c) => c,
  Bucket::Inf => break,
};
```

all the info i need in once place!